packages/driver/src/codecs/codecs.ts

```diff
@@ -49,8 +50,94 @@ import {
 } from "./pgvector";
 import { InternalClientError } from "../errors";
 
+import type { CodecContext } from "./context";
+
 import { INVALID_CODEC_ID, KNOWN_TYPENAMES, NULL_CODEC_ID } from "./consts";
 
+import type { Float16Array } from "../adapter.shared.node";
+
+// Types for Client.withCodecs() API:
+export namespace Codecs {
+  export type Codec<T> = {
+    encode: (data: any) => T;
+    decode: (data: T) => any;
+  };
+
+  export type AnyCodec = Codec<any>;
+
+  export type BoolCodec = Codec<boolean>;
+  export type Int16Codec = Codec<number>;
+  export type Int32Codec = Codec<number>;
+  export type Int64Codec = Codec<bigint>;
+  export type Float32Codec = Codec<number>;
+  export type Float64Codec = Codec<number>;
+  export type BigIntCodec = Codec<string>;
```

Comment:
```suggestion
  export type BigIntCodec = Codec<bigint>;
```

Reply:
Believe it or not -- this is intentional, as it's an extra step for us to create a bigint so string is more fundamental. So I was torn. That said it's highly unlikely someone brings their own bigint implementation

Reply:
Ah, ok. That makes sense.

Reply:
Should we also do that for Int64Codec?

Reply:
Int64Codec gives you bigint already as that's the most efficient way (we can read int64 to bigint in node fast).

I think I'll go and update the bigint codec to give you bigint. The point of this specific API isn't to replace native bigint with some stuff, but rather to wrap it in `Money` or something, for which us returning a string makes no sense.

So I'll incorporate James' suggestion.